test cases/java/11 shared library resource/src/com/mesonbuild/SharedLibraryResource.java

```diff
@@ -0,0 +1,41 @@
+package com.mesonbuild;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.IOException;
+
+public class SharedLibraryResource {
+    private static String libname = "nativecode";
+    static {
+        String fullname = System.mapLibraryName(libname);
+        // this is somewhat hacky, but attempt to split off the extension
+        int ext = fullname.indexOf(".");
```

Comment:
Java doesn't have a stem function for this?

Reply:
It seemingly doesn't. And even if it did, I'd consider a stem function is just as hacky. Ideally you could just access lib prefix and suffix from java, but you can't, so you have to gain access to the suffix for `mkstemp` somehow.

No idea if indexOf or lastIndexOf is better, the first breaks if the library prefix or libname contains a `.`, the second breaks if the extension were to contain a `.` (dunno, imagine a system capable of loading `.so.gz` or so)